### PR TITLE
enh(API): Interfaces for find security policy use case

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description
This PR add the interfaces structures and test for the use case find security policy.

**Fixes** # MON-11776

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * @var int|null
+     */
+    public ?int $passwordExpiration = null;
+
+    /**
+     * @var int|null
+     */
+    public ?int $delayBeforeNewPassword = null;
```
Comment (adr-mo) [inline src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php:76]: If I do remember we said that even if there is no parameters to the constructor we should add it anyway

File: src/Core/Domain/Security/Model/SecurityPolicy.php [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        private int $passwordMinimumLength,
+        private bool $hasUppercase,
+        private bool $hasLowercase,
+        private bool $hasNumber,
+        private bool $hasSpecialCharacter,
+        private bool $canReusePassword,
+        private ?int $attempts = null,
+        private ?int $blockingDuration = null,
+        private ?int $passwordExpiration = null,
+        private ?int $delayBeforeNewPassword = null
```
Comment (adr-mo) [inline src/Core/Domain/Security/Model/SecurityPolicy.php:65]: I guess if those can be null, it means that they are not mandatory to instanciate the class. So you can remove those from the constructor and add them as properties of your class with their getters and setters

File: src/Core/Domain/Security/Model/SecurityPolicy.php [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+     * @param int $passwordMinimumLength
+     * @param bool $hasUppercase
+     * @param bool $hasLowercase
+     * @param bool $hasNumber
+     * @param bool $hasSpecialCharacter
+     * @param bool $canReusePassword
+     * @param int|null $attempts
+     * @param int|null $blockingDuration
+     * @param int|null $passwordExpiration
+     * @param int|null $delayBeforeNewPassword
```
Comment (adr-mo) [inline src/Core/Domain/Security/Model/SecurityPolicy.php:53]: Missing phpDoc on potential Assertion exceptions

File: src/Core/Domain/Security/Model/SecurityPolicy.php (hunk 2 above)
Comment (jeremyjaouen) [reply src/Core/Domain/Security/Model/SecurityPolicy.php:65]: Agreed in many common case, but here in a business logic those properties are mandatory, even if they have a null value. this is why i add them to the constructor.

File: src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php (hunk 1 above)
Comment (jeremyjaouen) [reply src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php:76]: prove it ! 😄

File: src/Core/Infrastructure/Security/Api/FindSecurityPolicy/FindSecurityPolicyController.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+use Centreon\Application\Controller\AbstractController;
+use Core\Application\Security\UseCase\FindSecurityPolicy\FindSecurityPolicy;
+use Core\Application\Security\UseCase\FindSecurityPolicy\FindSecurityPolicyPresenterInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+class FindSecurityPolicyController extends AbstractController
+{
+    /**
+     * @param FindSecurityPolicy $useCase
+     * @param FindSecurityPolicyPresenterInterface $presenter
```
Comment (kduret) [inline src/Core/Infrastructure/Security/Api/FindSecurityPolicy/FindSecurityPolicyController.php:35]: missing phpdoc of return type

File: src/Core/Domain/Security/Model/SecurityPolicy.php [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+declare(strict_types=1);
+
+namespace Core\Domain\Security\Model;
+
+use Centreon\Domain\Common\Assertion\Assertion;
+use Centreon\Domain\Common\Assertion\AssertionException;
+
+class SecurityPolicy
+{
+
```
Comment (kduret) [inline src/Core/Domain/Security/Model/SecurityPolicy.php:31]:
```suggestion
```

File: src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * @var bool
+     */
+    public bool $canReusePassword;
+
+    /**
+     * @var int|null
+     */
+    public ?int $attempts = null;
```
Comment (kduret) [inline src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php:61]:
```suggestion
    public ?int $attempts;
```

File: src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * @var int|null
+     */
+    public ?int $blockingDuration = null;
```
Comment (kduret) [inline src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php:66]:
```suggestion
    public ?int $blockingDuration;
```

File: src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php (hunk 1 above)
Comment (kduret) [inline src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php:76]:
```suggestion
    public ?int $delayBeforeNewPassword;
```

File: src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * @var int|null
+     */
+    public ?int $passwordExpiration = null;
```
Comment (kduret) [inline src/Core/Application/Security/UseCase/FindSecurityPolicy/FindSecurityPolicyResponse.php:71]:
```suggestion
    public ?int $passwordExpiration;
```

**Review by kduret — APPROVED** [review]

**Review by AssanDiomande — APPROVED** [review]

**Pull request merged**